core/src/retransmit_stage.rs

```diff
@@ -526,6 +514,46 @@ impl RetransmitStage {
     }
 }
 
+impl AddAssign for RetransmitSlotStats {
+    fn add_assign(&mut self, other: Self) {
+        let Self {
+            num_shreds_received,
+            num_shreds_sent,
+        } = other;
+        for k in 0..3 {
+            self.num_shreds_received[k] += num_shreds_received[k];
+            self.num_shreds_sent[k] += num_shreds_sent[k];
+        }
+    }
+}
+
+impl RetransmitSlotStats {
+    fn submit(&self, slot: Slot) {
+        let num_shreds: usize = self.num_shreds_received.iter().sum();
+        let num_nodes: usize = self.num_shreds_sent.iter().sum();
+        datapoint_info!(
+            "retransmit-stage-slot-stats",
+            ("slot", slot, i64),
+            ("num_shreds", num_shreds, i64),
+            ("num_nodes", num_nodes, i64),
+            ("num_shreds_received_root", self.num_shreds_received[0], i64),
+            (
+                "num_shreds_received_1st_layer",
+                self.num_shreds_received[1],
+                i64
+            ),
+            (
+                "num_shreds_received_2nd_layer",
+                self.num_shreds_received[2],
+                i64
+            ),
+            ("num_shreds_sent_root", self.num_shreds_sent[0], i64),
+            ("num_shreds_sent_1st_layer", self.num_shreds_sent[1], i64),
+            ("num_shreds_sent_2nd_layer", self.num_shreds_sent[2], i64),
+        );
```

Comment:
Some timing metrics that might be useful to add here or in a follow-up PR:

1. The timestamp of the last shred that this validator retransmitted for this slot
2. Delay between first shred received for this slot and last shred that this validator retransmitted for this slot

The goal here being to see if the validator got the shred slow from its turbine parent

Also would be nice to have a way to measure if shreds for a slot sat in the retransmit queue for too long, but not sure of the best way to do this yet.

Reply:
done.

Note that timestamp is local node's wallclock time, and I have seen validator node's having clock drift. So it cannot be used for comparing across different nodes.
But I guess it will be useful to estimate latencies between different stages on the same node. e.g. window-service -> retransmit.